File: php/features.php
<?php
declare(strict_types=1);

// Techy SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class TechyFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TechyBaseFeature();
            case "test":
                return new TechyTestFeature();
            default:
                return new TechyBaseFeature();
        }
    }
}

File: py/features.py
# Techy SDK feature factory

from feature.base_feature import TechyBaseFeature
from feature.test_feature import TechyTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TechyBaseFeature(),
        "test": lambda: TechyTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
